powershell_kernel/powershell_proxy.py

```diff
@@ -92,8 +101,9 @@ def update_view_loop(self):
 
     def write(self, packet):
         # this is a hack to detect when we stop processing this input
-        if packet == '^':
+        if packet.endswith('^'):
```

Comment:
Why endswith here? Is there a chance we might get actual output that ends in this character?

Reply:
Yes, I used `endswith` due to the issue that currently exists when large packets need to be processed (for example a large string in a for loop) and the kernel hangs (and outputs ^ at the end) after the user tries to re-run the code cell multiple times. Since the long string example outputs the response by separating the message in different packets, it will cause the "^" to get attached to the end of the packet, as a result, using `endswith` solves the issue.